testsuite/classlibrary/TestCondVar.sc

```diff
@@ -0,0 +1,386 @@
+// Custom test class for checking exceptions thrown in predicates
+TestCondVarError : Error {}
+
+// Custom test classes to test for conversions to int and float
+TestCondVarBadIntConversion { asInteger { ^nil } }
+TestCondVarBadFloatConversion { asFloat { ^nil } }
+TestCondVarGoodIntConversion { asInteger { ^0 } }
+TestCondVarGoodFloatConversion { asFloat { ^0.0 } }
+
+TestCondVar : UnitTest {
+	// The test cases for `wait` cover situations with multiple threads, these are assumed
+	// to hold for `waitFor` as well.
+
+	// Tiny yields until a condition is true. Normally, we would use CondVar for this, but since we are testing CondVar
+	// itself, we don't want to rely on any of its functionality.
+	momentarilyYieldUntil { |predicate| while { predicate.value.not } { 0.yield } }
+
+	smallTimeSlice { ^1e-6 }
+
+	// wait - without predicate
+	test_wait_withoutPredicate_oneThread_signalOne {
+		var c = CondVar();
+		var waitResult, signalOneResult;
+		fork {
+			signalOneResult = c.signalOne;
+		};
+
+		// will timeout if signalOne does not wake us
+		waitResult = c.wait;
```

Comment:
Nothing will cause this test to timeout. The test will hang indefinitely if `c.signalOne` happens just before `c.wait`. One way to fix this would be to schedule `c.signalOne` some small amount of time into the future.

Reply:
This isn't something I'm very knowledgeable in. Would adding this to the fork be an improvement?
```
fork {
    thisThread.clock.sched(this.smallTimeSlice, { signalOneResult = c.signalOne });
};
```

Reply:
I don't want to keep this PR from being merged. I'm resolving this conversation.